zerver/webhooks/newrelic/view.py

```diff
@@ -1,169 +1,264 @@
+        additional_info_block = generate_additional_info_message(payload)
+        if additional_info_block:
+            body_detail += additional_info_block
 
-        check_send_webhook_message(request, user_profile, topic_name, content, info["status"])
-        return json_success(request)
+        custom_payload_block = generate_custom_field_message(payload)
+        if custom_payload_block:
+            body_detail += custom_payload_block
 
-    # Handle new format
+        body = BASE_NOTIFICATION_WRAP_TEMPLATE.format(header=NOTIFICATION_HEADER, body=body_detail)
+        body = body.format(**message_context).strip()
+        topic = TOPIC_TEMPLATE.format(
+            state=message_context["state"], priority=message_context["priority"]
+        ).title()
+        check_send_webhook_message(request, user_profile, topic, body, message_context["state"])
     else:
-        info = {
-            "condition_name": payload.get("condition_name", "Unknown condition").tame(check_string),
-            "details": payload.get("details", "No details.").tame(check_string),
-            "incident_url": payload.get("issueUrl", "https://alerts.newrelic.com").tame(
-                check_string
-            ),
-            "incident_acknowledge_url": payload.get(
-                "incident_acknowledge_url", "https://alerts.newrelic.com"
-            ).tame(check_string),
-            "status": payload.get("state", "None").tame(check_string),
-            "iso_timestamp": "",
-            "owner": payload.get("owner", "").tame(check_string),
-        }
-
-        unix_time = payload.get("createdAt").tame(
-            check_none_or(check_union([check_string, check_int]))
+        # Fallback notification
+        body = FALL_BACK_NOTIFICATION_TEMPLATE.format(
+            integration_last_updated=INTEGRATION_LAST_UPDATED
         )
-        if unix_time is None:
-            raise JsonableError(_("The newrelic webhook requires timestamp in milliseconds"))
-
-        info["iso_timestamp"] = str(unix_milliseconds_to_timestamp(unix_time, "newrelic"))
-
-        # Add formatting to the owner field if owner is present
-        if info["owner"] != "":
-            info["owner"] = "by **{}** ".format(info["owner"])
-
-        # These are the three promised state values
-        if info["status"].lower() == "activated":
-            content = ACTIVE_TEMPLATE.format(**info)
-        elif info["status"].lower() == "acknowledged":
-            content = DEFAULT_TEMPLATE.format(**info)
-        elif info["status"].lower() == "closed":
-            content = DEFAULT_TEMPLATE.format(**info)
-        elif info["status"].lower() == "created":
-            content = DEFAULT_TEMPLATE.format(**info)
-        else:
-            raise JsonableError(
-                _(
-                    "The newrelic webhook requires state be in [created|activated|acknowledged|closed]"
-                )
-            )
+        topic = FALL_BACK_TOPIC_TEMPLATE
```

Comment:
This is used only here, so instead of using a constant, we could just have the string here.